### PR TITLE
Improve discoverability on Hugging Face

### PR DESCRIPTION
**Pull request by NielsRogge**
Hi @mv-lab and team,

Thanks for this nice work! 

I wrote a quick PoC to showcase that you can easily have integration with the 🤗 hub so that you can automatically:
* load the various DarkIR models using `from_pretrained` (and push them using `push_to_hub`
* track download numbers for your models (similar to models in the Transformers library)
* leverage [safetensors](https://github.com/huggingface/safetensors) for weights serialization
* have nice model cards on a per-model basis. It leverages the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which allows to inherits these methods.

Usage is as follows:

```python
from archs import DarkIR

model = DarkIR.from_pretrained("cidautai/darkir")
```

One can push a trained model as follows:

```python
from archs import DarkIR

model = DarkIR()
state_dict = ...
model.load_state_dict(state_dict)

#one can push a trained model to the hub
model.push_to_hub("cidautai/darkir-example")
```

This means people don't need to manually download a checkpoint first in their local environment, it just loads automatically from the hub.

Would you be interested in this integration?

Kind regards,

Niels

### CODE REVIEW

**Comment by PCMCI-Tonight** [comment]
Hi， As you take distributed training, so I am reporting error when I use single threaded training, I took some methods to change to single threaded but still reporting error, would like to ask any solution? Thank you
The image shows my changes

image
image
image

**Comment by ai4niuniu** [comment]
Hello, I would like to ask where the training code is. I want to train my own dataset. Thank you.

**Comment by Yolo1-gguo** [comment]
Dear author, hello! I have a question to ask you: How do you test the indicators such as MUSIQ, NRQM and NIQE? Is it dependent on the pyiqa library

**Comment by Yolo1-gguo** [comment]
Does anyone know which profile to read if you want to test the Real-LOLBlur dataset?

**Comment by wang-yu-tao** [comment]
> Hi， As you take distributed training, so I am reporting error when I use single threaded training, I took some methods to change to single threaded but still reporting error, would like to ask any solution? Thank you嗨，由于您正在进行分布式训练，所以我在使用单线程训练时报告了错误，我尝试了一些方法来改为单线程，但仍然报告错误，想问是否有任何解决方案？谢谢 The image shows my changes图片显示了我所做的更改
> 
> image image image



> Hi， As you take distributed training, so I am reporting error when I use single threaded training, I took some methods to change to single threaded but still reporting error, would like to ask any solution? Thank you嗨，由于您正在进行分布式训练，所以我在使用单线程训练时报告了错误，我尝试了一些方法来改为单线程，但仍然报告错误，想问是否有任何解决方案？谢谢 The image shows my changes图片显示了我所做的更改
> 
> image image image
作者用的是多卡分布式，如果只有单卡，需要进行修改相应代码，我改了一天的测试和推理以适应单卡。但是作者没有提供训练相关的文件和教程，就比较麻烦，尝试自己写了一天最终放弃了，希望作者可以上传一下吧。

**Comment by wang-yu-tao** [comment]
Is it convenient for the author to supplement the training code and usage guide in the project? If we could share the relevant resources, it would be of great help to our subsequent research. Thank you in advance!

**Comment by dhhdhdwudhx** [comment]
啥玩意



Sent&nbsp;from&nbsp;my&nbsp;iPhone


------------------ Original ------------------
From: wang-yu-tao ***@***.***&gt;
Date: Tue,Jul 29,2025 4:16 PM
To: cidautai/DarkIR ***@***.***&gt;
Cc: dhhdhdwudhx ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [cidautai/DarkIR] Improve discoverability on Hugging Face (PR #4)



wang-yu-tao left a comment (cidautai/DarkIR#4)
  
Hello, I would like to ask where the training code is. I want to train my own dataset. Thank you.你好，我想问一下训练代码在哪里。我想训练自己的数据集。谢谢。
  
您好，不知道您改成功没有，我也需要这个。

**Comment by timeproud** [comment]
> Is it convenient for the author to supplement the training code and usage guide in the project? If we could share the relevant resources, it would be of great help to our subsequent research. Thank you in advance!

Hello, have you obtained the training code?

**Comment by NielsRogge** [comment]
Looks like a model has been uploaded by @danifei: https://huggingface.co/Cidaut/DarkIR

Would be great to mention the 🤗 model in the README of this Github repo :)

**Comment by potender** [comment]
Dear author, thank you for your work. I've incorporated EBlock and DBlock into other networks, and the resulting loss function is NaN, with the images getting progressively worse until they become completely black. Do you have any experience with this?

**Comment by gyniy** [comment]
Hello, I would like to ask where the training code is. I want to train my own dataset. Thank you.

**Comment by yixuang615-rgb** [comment]
> 您好，请问训练代码在哪里？我想用自己的数据集进行训练。谢谢。

你好，请问有训练代码了吗？

**Comment by yixuang615-rgb** [comment]
> > 作者是否方便在项目中补充训练代码和使用指南？如果能分享相关资源，将对我们后续的研究大有帮助。提前感谢！
> 
> 你好，请问你拿到培训码了吗？

你好，请问有训练代码了吗？

**Comment by XuLi666231** [comment]
Does anyone have the training code? Thank you.

**Comment by mengyingjidu** [comment]
请问一下训练代码去哪里找有找到的滴滴我谢谢
